File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/ItemWithNotes.java
package io.zeitwert.fm.collaboration.model;

// This interface defines note operations that are implemented by AggregateWithNotesMixin.
import java.util.List;
import io.zeitwert.fm.collaboration.model.enums.CodeNoteType;

public interface ItemWithNotes {

	List<ObjNote> getNotes();

	ObjNote addNote(CodeNoteType noteType);

	void removeNote(Object noteId);

}

File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/ItemWithNotes.kt
package io.zeitwert.fm.collaboration.model

import io.zeitwert.fm.collaboration.model.enums.CodeNoteType

interface ItemWithNotes {

	val notes: List<Any>

	fun addNote(
		noteType: CodeNoteType,
		userId: Any,
	): ObjNote

	fun removeNote(
		noteId: Any,
		userId: Any,
	)

}
